File: SEGUNDO PERIODO/AEDS II/TPS/02/Digitos.java
import java.util.Scanner;

public class Digitos {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // enquanto houver inteiros na entrada
        while (sc.hasNextInt()) {
            int numero = sc.nextInt();
            if (numero == 0)
                break;

            int resultado = Soma(numero);
            System.out.println(resultado);
        }

        sc.close();
    }

    // método que soma os dígitos de um número positivo
    public static int Soma(int num) {
        int soma = 0;
        while (num != 0) {
            soma += num % 10; // pega o último dígito
            num /= 10; // remove o último dígito
        }
        return soma;
    }
}

File: SEGUNDO PERIODO/AEDS II/TPS/02/Digitos.c
#include <stdio.h>

// função recursiva que soma os dígitos
int somaDigitos(int n)
{
    if (n == 0)
        return 0; // caso base
    return (n % 10) + somaDigitos(n / 10);
}

int main()
{
    int num;

    // enquanto conseguir ler um inteiro
    while (scanf("%d", &num) == 1)
    {
        printf("%d\n", somaDigitos(num));
    }

    return 0;
}
